src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -0,0 +1,392 @@
+package org.broadinstitute.hellbender.utils.pileup;
+
+import com.google.api.client.repackaged.com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.fragments.FragmentCollection;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a pileup of reads at a given position.
+ */
+public final class ReadPileup implements Iterable<PileupElement>{
+    private final Locatable loc;
+    private final List<PileupElement> pileupElements;
+
+    /**
+     * Create a new version of a read backed pileup at loc, using the reads and their corresponding
+     * offsets.
+     * Note: This internal-use constructor keeps an alias pointer to the given list.
```

Comment:
Also, the constructor description is misleading and implies that a copy is being made ("Create a new version of a read backed pileup at loc") when in fact no copy is performed and the parameters are just directly assigned to fields.